### PR TITLE
Create Helm chart compatibility reference for Software

### PR DESCRIPTION
**Pull request by jwitz**
Resolves https://github.com/astronomer/docs/issues/3222
Resolves https://github.com/astronomer/docs/issues/3192

### CODE REVIEW

File: software/version-compatibility-reference.md [hunk 1]
```diff
@@ -56,6 +56,36 @@ For more information on upgrading Kubernetes versions, follow the guidelines off
 - [Google GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades)
 - [RedHat OpenShift](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/updating_clusters/index)
 
+## Airflow chart compatibility reference
+
+Each version of Astronomer Software specifies a version of the [Airflow Helm chart](https://airflow.apache.org/docs/helm-chart/stable/index.html) to use for creating new Airflow Deployments. A Deployment's Airflow chart defines how a Deployment interacts autoscales Pods and interacts with other components in your cluster.
```
Comment (jwitz) [inline software/version-compatibility-reference.md:61]: Why do customers traditionally care about their Helm chart version? Is there something we can add here about why your chart version is important that's specific to Astronomer Software?

**Review by lzdanski — APPROVED** [review]
Content looks good in preview. Good to merge after technical review.

File: software/version-compatibility-reference.md (hunk 1 above)
Comment (danielhoherd) [inline software/version-compatibility-reference.md:61]: > specifies a version of the [Airflow Helm chart](https://airflow.apache.org/docs/helm-chart/stable/index.html)

This is not accurate. I'm not sure what the goal of this section is so I can't suggest improvements, but I can explain the relationships.

The astronomer/astronomer helm chart references the astronomer/airflow-chart version. The astronomer/airflow-chart version references the apache/airflow chart version. So for instance, in the table below, 1.9.5 is a reference to https://github.com/astronomer/airflow-chart/tree/v1.9.5 and not https://github.com/apache/airflow/releases/tag/helm-chart/1.9.0 (helm chart tag 1.9.5 actually doesn't exist in apache/airflow)

Comment (jwitz) [inline software/version-compatibility-reference.md:61]:
```suggestion
Each version of Astronomer Software specifies a version of the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart), which is used to create new Airflow Deployments. A Deployment's Airflow chart defines how a Deployment interacts autoscales Pods and interacts with other components in your cluster. 
```

File: software/version-compatibility-reference.md [hunk 2]
```diff
@@ -56,6 +56,36 @@ For more information on upgrading Kubernetes versions, follow the guidelines off
 - [Google GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades)
 - [RedHat OpenShift](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/updating_clusters/index)
 
+## Airflow chart compatibility reference
+
+Each version of Astronomer Software specifies a version of the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart), which is used to create new Airflow Deployments. A Deployment's Airflow chart defines how a Deployment interacts autoscales Pods and interacts with other components in your cluster. 
+
+Use the following table to see the Airflow Helm chart version for each supported version of Astronomer Software. To view the Airflow Helm chart for an unsupported version of Astronomer Software, open the default Astronomer Helm chart in the [`astronomer/astronomer` repository](https://github.com/astronomer/astronomer/blob/master/charts/astronomer/values.yaml), select the **Tag** for your desired version, and find the value for `airflowChartVersion` in the version's Helm chart.
+
+| Astronomer platform version | Airflow Helm chart version |
```
Comment (jwitz) [inline software/version-compatibility-reference.md:65]:
```suggestion
| Astronomer platform version | Astronomer Airflow Helm chart version |
```

File: software/version-compatibility-reference.md [hunk 3]
```diff
@@ -56,6 +56,36 @@ For more information on upgrading Kubernetes versions, follow the guidelines off
 - [Google GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades)
 - [RedHat OpenShift](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/updating_clusters/index)
 
+## Airflow chart compatibility reference
+
+Each version of Astronomer Software specifies a version of the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart), which is used to create new Airflow Deployments. A Deployment's Airflow chart defines how a Deployment interacts autoscales Pods and interacts with other components in your cluster.
```
Comment (jwitz) [inline software/version-compatibility-reference.md:61]:
```suggestion
Each version of Astronomer Software specifies a version of the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart). A Deployment's Airflow chart defines how a Deployment autoscales Pods and interacts with other components in your cluster. 
```

File: software/version-compatibility-reference.md [hunk 4]
```diff
@@ -56,6 +56,36 @@ For more information on upgrading Kubernetes versions, follow the guidelines off
 - [Google GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades)
 - [RedHat OpenShift](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/updating_clusters/index)
 
+## Airflow chart compatibility reference
+
+Each version of Astronomer Software specifies a version of the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart). A Deployment's Airflow chart defines how a Deployment autoscales Pods and interacts with other components in your cluster. 
+
+Use the following table to see the Airflow Helm chart version for each supported version of Astronomer Software. To view the Airflow Helm chart for an unsupported version of Astronomer Software, open the default Astronomer Helm chart in the [`astronomer/astronomer` repository](https://github.com/astronomer/astronomer/blob/master/charts/astronomer/values.yaml), select the **Tag** for your desired version, and find the value for `airflowChartVersion` in the version's Helm chart.
```
Comment (jwitz) [inline software/version-compatibility-reference.md:63]:
```suggestion
Use the following table to see the Airflow Helm chart version for each supported version of Astronomer Software. To view the Airflow Helm chart for an unsupported version of Astronomer Software, open the default Astronomer Helm chart in the [`astronomer/astronomer` repository](https://github.com/astronomer/astronomer/blob/master/charts/astronomer/values.yaml), select the **Tag** for your desired version, and review the value for `airflowChartVersion` in the version's Helm chart.
```

File: software/version-compatibility-reference.md [hunk 5]
```diff
@@ -56,6 +56,36 @@ For more information on upgrading Kubernetes versions, follow the guidelines off
...
+| 0.31.2                      | 1.7.6                      |
+| 0.32.2                      | 1.8.7                      |
+| 0.32.3                      | 1.8.8                      |
+| 0.32.4                      | 1.8.8                      |
+| 0.32.5                      | 1.8.9                      |
+| 0.33.0                      | 1.9.2                      |
+| 0.33.1                      | 1.9.4                      |
+| 0.33.2                      | 1.9.5                      |
+
+
```
Comment (jwitz) [inline software/version-compatibility-reference.md:87]:
```suggestion
```

File: software/version-compatibility-reference.md [hunk 6]
```diff
@@ -56,6 +56,34 @@ For more information on upgrading Kubernetes versions, follow the guidelines off
 - [Google GKE](https://cloud.google.com/kubernetes-engine/docs/concepts/cluster-upgrades)
 - [RedHat OpenShift](https://access.redhat.com/documentation/en-us/openshift_container_platform/4.11/html/updating_clusters/index)
 
+## Airflow chart compatibility reference
+
+Each version of Astronomer Software specifies a version of the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart). A Deployment's Airflow chart defines how a Deployment autoscales Pods and interacts with other components in your cluster.
```
Comment (jwitz) [inline software/version-compatibility-reference.md:61]:
```suggestion
Astronomer Software clusters utilize the [Astronomer-distributed Helm chart for Apache Airflow](https://github.com/astronomer/airflow-chart). A Deployment's Airflow chart defines how a Deployment autoscales Pods and interacts with other components in your cluster. 
```

**Pull request merged**